app/assets/javascripts/angular/controllers/messages.js

```diff
@@ -0,0 +1,107 @@
+angular.module('messagesApp', ['truncate']).controller('messagesCtrl', ['$scope', '$element', '$window', function($scope, $element, $window) {
+  $scope.myCountry = "Germany";
+
+  // Tracking orientation of the phone.
+  $scope.getWindowOrientation = function () {
+    return $window.orientation;
+  };
+
+  $scope.$watch($scope.getWindowOrientation, function (newValue, oldValue) {
+    $scope.orientation = newValue === 0 ? 'portrait' : 'landscape';
+  }, true);
+
+  angular.element($window).bind('orientationchange', function () {
+    $scope.$apply();
+  });
+
+  // Data should be grouped by who the conversation is with.
+  $scope.myMessages = [
```

Comment:
Here's the data structure that I mocked up for messages. The messages themselves haven't changed, but what I've done is grouped them together into "conversations". All that it actually adds is a property for a) who the conversation is actually with, and b) when the latest message was sent. But the grouping will be necessary to make this entire thing work, and those two properties just make some things easier on my end.